File: system-test/java/src/main/java/com/mycompany/myshop/testkit/driver/port/MyShopDriver.java
package com.mycompany.myshop.testkit.driver.port;

import com.mycompany.myshop.testkit.driver.port.dtos.error.SystemError;
import com.mycompany.myshop.testkit.driver.port.dtos.BrowseCouponsRequest;
import com.mycompany.myshop.testkit.driver.port.dtos.BrowseCouponsResponse;
import com.mycompany.myshop.testkit.driver.port.dtos.CancelOrderRequest;
import com.mycompany.myshop.testkit.driver.port.dtos.CancelOrderResponse;
import com.mycompany.myshop.testkit.driver.port.dtos.DeliverOrderRequest;
import com.mycompany.myshop.testkit.driver.port.dtos.DeliverOrderResponse;
import com.mycompany.myshop.testkit.driver.port.dtos.GoToMyShopRequest;
import com.mycompany.myshop.testkit.driver.port.dtos.GoToMyShopResponse;
import com.mycompany.myshop.testkit.driver.port.dtos.PlaceOrderRequest;
import com.mycompany.myshop.testkit.driver.port.dtos.PlaceOrderResponse;
import com.mycompany.myshop.testkit.driver.port.dtos.PublishCouponRequest;
import com.mycompany.myshop.testkit.driver.port.dtos.PublishCouponResponse;
import com.mycompany.myshop.testkit.driver.port.dtos.ViewOrderRequest;
import com.mycompany.myshop.testkit.driver.port.dtos.ViewOrderResponse;
import com.mycompany.myshop.testkit.common.Result;

public interface MyShopDriver extends AutoCloseable {
    Result<GoToMyShopResponse, SystemError> goToMyShop(GoToMyShopRequest request);

    Result<PlaceOrderResponse, SystemError> placeOrder(PlaceOrderRequest request);

    Result<CancelOrderResponse, SystemError> cancelOrder(CancelOrderRequest request);

    Result<DeliverOrderResponse, SystemError> deliverOrder(DeliverOrderRequest request);

    Result<ViewOrderResponse, SystemError> viewOrder(ViewOrderRequest request);

    Result<PublishCouponResponse, SystemError> publishCoupon(PublishCouponRequest request);

    Result<BrowseCouponsResponse, SystemError> browseCoupons(BrowseCouponsRequest request);
}

File: system-test/typescript/src/testkit/driver/port/my-shop-driver.ts
import { Result } from '../../common/result.js';
import { AsyncCloseable } from './async-closeable.js';
import { GoToMyShopRequest } from './dtos/GoToMyShopRequest.js';
import { GoToMyShopResponse } from './dtos/GoToMyShopResponse.js';
import { PlaceOrderRequest } from './dtos/PlaceOrderRequest.js';
import { PlaceOrderResponse } from './dtos/PlaceOrderResponse.js';
import { CancelOrderRequest } from './dtos/CancelOrderRequest.js';
import { CancelOrderResponse } from './dtos/CancelOrderResponse.js';
import { DeliverOrderRequest } from './dtos/DeliverOrderRequest.js';
import { DeliverOrderResponse } from './dtos/DeliverOrderResponse.js';
import { ViewOrderRequest } from './dtos/ViewOrderRequest.js';
import { ViewOrderResponse } from './dtos/ViewOrderResponse.js';
import { SystemError } from './dtos/errors/SystemError.js';
import { PublishCouponRequest } from './dtos/PublishCouponRequest.js';
import { PublishCouponResponse } from './dtos/PublishCouponResponse.js';
import { BrowseCouponsRequest } from './dtos/BrowseCouponsRequest.js';
import { BrowseCouponsResponse } from './dtos/BrowseCouponsResponse.js';

export interface MyShopDriver extends AsyncCloseable {
  goToMyShop(request: GoToMyShopRequest): Promise<Result<GoToMyShopResponse, SystemError>>;
  placeOrder(request: PlaceOrderRequest): Promise<Result<PlaceOrderResponse, SystemError>>;
  cancelOrder(request: CancelOrderRequest): Promise<Result<CancelOrderResponse, SystemError>>;
  deliverOrder(request: DeliverOrderRequest): Promise<Result<DeliverOrderResponse, SystemError>>;
  viewOrder(request: ViewOrderRequest): Promise<Result<ViewOrderResponse, SystemError>>;
  publishCoupon(request: PublishCouponRequest): Promise<Result<PublishCouponResponse, SystemError>>;
  browseCoupons(request: BrowseCouponsRequest): Promise<Result<BrowseCouponsResponse, SystemError>>;
}
